File: Adm/Models/src/Combinacao.java
package Adm.Models.src;

import java.util.HashMap;
import java.util.Map;

public class Combinacao implements Inter{
  private int id;
  private String clima;
  private int id_itens_roupas;

  public Combinacao(int id, String clima, int id_itens_roupas) {
    this.setId(id);
    this.setClima(clima);
    this.setIdItens_roupas(id_itens_roupas);
  }

  @Override
  public String toString() {
    return String.format("Combinação:" +
              "id=" + this.id +
              ", clima=" + this.clima +
              ", id_itens=" + this.id_itens_roupas);
  }

  public Map<String, Object> toDict() {
    Map<String, Object> dict = new HashMap<>();
    dict.put("id", this.id);
    dict.put("clima", this.clima);
    dict.put("id_itens", this.id_itens_roupas);
    return dict;
}

  public void setId(int id) {
    if (id >= 0) {
      this.id = id;
    } else {
      throw new IllegalArgumentException("ID combinação inválido");
    }
  }

  public void setClima(String clima) {
    if (clima != null) {
      this.clima = clima;
    } else {
      throw new IllegalArgumentException("clima combinação inválido");
    }
  }

  public void setIdItens_roupas(int id_itens_roupas) {
    if (id_itens_roupas >= 0) {
      this.id_itens_roupas = id_itens_roupas;
    } else {
      throw new IllegalArgumentException("id item_roupa area combinação inválida");
    }
  }

  public int getId() {
    return this.id;
  }

  public String getClima() {
    return this.clima;
  }

  public int getIdItens_roupas() {
    return this.id_itens_roupas;
  }
}

File: Adm/Models/Combinacao.cs
using System;
using System.Collections.Generic;

namespace Adm.Models
{
    public class Combinacao : Inter
    {
        // Atributos privados
        private int id;
        private string clima;
        private int id_itens;

        public Combinacao(int id, string clima, int id_itens)
        {
            setId(id);
            setClima(clima);
            set_idItens(id_itens);
        }

        // Métodos setters com validação
        public void setId(int id)
        {
            if (id < 0)
            {
                throw new ArgumentException("ID inválido");
            }
            this.id = id;
        }

        public void setClima(string clima)
        {
            if (string.IsNullOrWhiteSpace(clima))
            {
                throw new ArgumentException("clima não pode ser vazio");
            }
            this.clima = clima;
        }

        public void set_idItens(int id_itens)
        {
            if (id_itens < 0)
            {
                throw new ArgumentException("ID itens inválido");
            }
            this.id_itens = id_itens;
        }

     
        // Métodos getters
        public int getId() => this.id;
        public string getClima() => this.clima;
        public int get_idItens() => this.id_itens;
        
        public Dictionary<string, object> ToDict()
        {
            return new Dictionary<string, object>
            {
                { "id", this.id },
                { "clima", this.clima},
                { "id_itens", this.id_itens }
               };
            }

        // Método ToString
        public override string ToString()
        {
            return $"Cliente: id={this.id}, clima={this.clima}, id_itens={this.id_itens}";
        }
    }
}
        
